src/Query/DataReaderInterface.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Db\Query;
+
+use Closure;
+use Countable;
+use Iterator;
+
+/**
+ * This interface represents a forward-only stream of rows from a query result set.
+ *
+ * To read the current row of data just iterate on it.
+ *
+ * For example,
+ *
+ * ```php
+ * $command = $connection->createCommand('SELECT * FROM post');
+ * $reader = $command->query();
+ *
+ * foreach ($reader as $row) {
+ *     $rows[] = $row;
+ * }
+ *
+ * Note: That since DataReader is a forward-only stream, you can only traverse it once. Doing it the second time will
+ * throw an exception.
+ *
+ * @extends Iterator<int|string|null, array|false>
+ *
+ * @psalm-import-type IndexBy from QueryPartsInterface
+ */
+interface DataReaderInterface extends Iterator, Countable
```

Comment:
Consider reusing interfaces from https://github.com/yiisoft/data/blob/master/src/Reader/DataReaderInterface.php. Not mandatory though...

Reply:
They have the same name but different meaning
- `Data\Reader\DataReaderInterface` is for reading from `QueryInterface`
- `Db\Query\DataReaderInterface` is for reading from `PDOStatement`